tools/dds/dds-adapter/lrs-device-controller.cpp

```diff
@@ -656,3 +658,68 @@ float lrs_device_controller::query_option( const std::shared_ptr< realdds::dds_o
     rs2_option opt_type = option_name_to_type( option->get_name(), sensor );
     return sensor.get_option( opt_type );
 }
+
+
+void lrs_device_controller::override_default_profiles( const std::map< std::string, realdds::dds_stream_profiles > & stream_name_to_profiles,
+                                                       std::map< std::string, size_t > & stream_name_to_default_profile ) const
+{
+    static const std::string RS410_PID = "0AD2";
+    static const std::string RS415_PID = "0AD3";
+
+    std::string product_line = _rs_dev.get_info( RS2_CAMERA_INFO_PRODUCT_LINE );
+    std::string product_id = _rs_dev.get_info( RS2_CAMERA_INFO_PRODUCT_ID );
+
+    if( product_line == "D400" )
+    {
+        uint16_t fps = 30;
+        uint16_t width = 848;
+        uint16_t height = 480;
+        if( product_id == RS415_PID || product_id == RS410_PID )
+        {
+            width = 1280;
+            height = 720;
+        }
+        stream_name_to_default_profile["Depth"] = get_index_of_profile( stream_name_to_profiles.at( "Depth" ),
+            realdds::dds_video_stream_profile( fps, realdds::dds_stream_format::from_rs2( RS2_FORMAT_Z16 ), width, height ) );
+        stream_name_to_default_profile["Infrared_1"] = get_index_of_profile( stream_name_to_profiles.at( "Infrared_1" ),
+            realdds::dds_video_stream_profile( fps, realdds::dds_stream_format::from_rs2( RS2_FORMAT_Y8 ), width, height ) );
+        stream_name_to_default_profile["Infrared_2"] = get_index_of_profile( stream_name_to_profiles.at( "Infrared_2" ),
+            realdds::dds_video_stream_profile( fps, realdds::dds_stream_format::from_rs2( RS2_FORMAT_Y8 ), width, height ) );
+
+        width = 1280;
+        height = 720;
+        stream_name_to_default_profile["Color"] = get_index_of_profile( stream_name_to_profiles.at( "Color" ),
+            realdds::dds_video_stream_profile( fps, realdds::dds_stream_format::from_rs2( RS2_FORMAT_YUYV ), width, height ) );
+    }
+}
+
+
+size_t lrs_device_controller::get_index_of_profile( const realdds::dds_stream_profiles & profiles,
```

Comment:
This looks like a subset of `find_profile()` in this same file...

Reply:
It is similar and at first I tried to use it, but it was just bending the code to use it too much.
`find_profile` searches in a stream_server profiles list and return a profile object, I needed to get the index of a profile in another list. It was easier and clearer to write a new function.

I was more inclined to use `profiles_are_compatible` in my new function, but didn't want to create `shared_ptr`s just for it, the allocation and dynamic_cast are unneeded overhead.